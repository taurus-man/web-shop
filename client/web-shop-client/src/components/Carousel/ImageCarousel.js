"use client"
import Image from 'next/image';
import { useEffect, useState } from 'react';

export default function ImageCarousel() {
    const images = ['https://picsum.photos/seed/random101/400/500', 'https://picsum.photos/seed/random102/400/500', 'https://picsum.photos/seed/random103/400/500'];
    const [currentIndex, setCurrentIndex] = useState(0);

    useEffect(() => {
        const intervalId = setInterval(() => {
            setCurrentIndex((currentIndex) => (currentIndex + 1) % images.length);
        }, 5000);

        return () => clearInterval(intervalId);
    }, [images.length]);

    return (
        <div className="flex flex-col items-center relative">
            <div className='w-36 h-64 relative lg:w-full'>
                <Image
                    src={images[currentIndex]}
                    alt="Featured"
                    fill
                    style={{ objectFit: 'cover' }}
                    sizes="(max-width: 768px) 100vw, (max-width: 1024px) 50vw, 25vw"
                    priority={true}
                />
            </div>
            <div className="flex relative z-50 justify-center space-x-1 mt-2">
                {images.map((_, index) => (
                    <div
                        key={index}
                        className={`block h-2 w-2 rounded-full ${index === currentIndex ? 'bg-red-600' : 'bg-gray-300'}`}
                        onClick={() => setCurrentIndex(index)}
                    />
                ))}
            </div>
            <div className='mt-2 mb-2 border-b-2 w-full'>
                <span className="text-xs font-bold uppercase">Featured Post</span>
            </div>
        </div>
    );
}
