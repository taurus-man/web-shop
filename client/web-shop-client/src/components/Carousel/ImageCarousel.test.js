import { render } from '@testing-library/react'
import { useRouter } from 'next/navigation'
import ImageCarousel from './ImageCarousel';

global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () => Promise.resolve({ rates: {id: 1}}),
  })
);

beforeEach(() => {
  fetch.mockClear();
});

jest.spyOn(console, 'error').mockImplementation(jest.fn());

// Mock useRouter:
jest.mock("next/navigation", () => ({
    useRouter() {
      return {
        prefetch: () => null
      };
    }
  }));

 
it('renders correctly', () => {
  const { container } = render(<ImageCarousel />)
  expect(container).toMatchSnapshot()
})
