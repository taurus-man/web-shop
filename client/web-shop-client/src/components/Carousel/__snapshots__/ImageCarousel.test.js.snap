// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="flex flex-col items-center relative"
  >
    <div
      class="w-36 h-64 relative lg:w-full"
    >
      <img
        alt="Featured"
        data-nimg="fill"
        decoding="async"
        fetchpriority="high"
        sizes="(max-width: 768px) 100vw, (max-width: 1024px) 50vw, 25vw"
        src="/_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Frandom101%2F400%2F500&w=3840&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Frandom101%2F400%2F500&w=256&q=75 256w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Frandom101%2F400%2F500&w=384&q=75 384w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Frandom101%2F400%2F500&w=640&q=75 640w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Frandom101%2F400%2F500&w=750&q=75 750w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Frandom101%2F400%2F500&w=828&q=75 828w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Frandom101%2F400%2F500&w=1080&q=75 1080w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Frandom101%2F400%2F500&w=1200&q=75 1200w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Frandom101%2F400%2F500&w=1920&q=75 1920w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Frandom101%2F400%2F500&w=2048&q=75 2048w, /_next/image?url=https%3A%2F%2Fpicsum.photos%2Fseed%2Frandom101%2F400%2F500&w=3840&q=75 3840w"
        style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: cover; color: transparent;"
      />
    </div>
    <div
      class="flex relative z-50 justify-center space-x-1 mt-2"
    >
      <div
        class="block h-2 w-2 rounded-full bg-red-600"
      />
      <div
        class="block h-2 w-2 rounded-full bg-gray-300"
      />
      <div
        class="block h-2 w-2 rounded-full bg-gray-300"
      />
    </div>
    <div
      class="mt-2 mb-2 border-b-2 w-full"
    >
      <span
        class="text-xs font-bold uppercase"
      >
        Featured Post
      </span>
    </div>
  </div>
</div>
`;
